backend/models/dynamodb/Content.js

```diff
@@ -0,0 +1,139 @@
+
+import DynamoDB from "@klaudsol/commons/lib/DynamoDB"
+import { 
+  DYNAMO_DB_TYPES, 
+  DYNAMO_DB_INDEXES, 
+  DYNAMO_DB_TABLE, 
+} from "@/constants";
+import RecordNotFound from "@klaudsol/commons/errors/RecordNotFound";
+import { formatAttributesData, formatEntityTypeSlugResponse, getEntityVariant, getOrder } from "@/utils/dynamodb/formatResponse";
+
+export default class Content {
+  // Initialize DB
+  static db = new DynamoDB();
+
+  // Gets all items under the content_type_slug
+  // Sample: api/tech
+  // Gets the content_type first based on its slug
+  // Returns the PK and other needed attributes
+  // Uses the PK to get all the items
+  // TODO: Add pagination
+  // TODO: Add filtering
+  // TODO: Add search
+  static async whereContentTypeSlug({ content_type_slug, order = null }) {
+    // Gets all the content_type info based on the content_type_slug
+    const { 
+      contentTypePK, 
+      contentTypeId, 
+      contentTypeAttributes, 
+      contentTypeVariant 
+    } = await this.whereContentType({ content_type_slug });
+            
+    const params = {
+      TableName: DYNAMO_DB_TABLE,
+      KeyConditionExpression: "#PK = :PK",
+      FilterExpression: "#type = :type",
```

Comment:
Filter is a `SCAN` operation, which reads every result. For large result sets, this is costly. I suggest we use `KeyConditionExpression` for this, with 2 lookups.

Reply:
I'll spend time modifying this branch. To reduce cost, we need to focus on querying keys only via `KeyConditionExpression`.

Reply:
Addressed sir @ardeearam